File: desktop/csharp/ChimeraLocale.cs
namespace Chimera.Desktop;
public enum ChimeraDirection { Ltr, Rtl }
public readonly record struct ChimeraLocale(string Id, string Language, string Script, ChimeraDirection Direction) {
  public static ChimeraLocale For(string? id) => id?.StartsWith("ar", System.StringComparison.OrdinalIgnoreCase) == true ? new("ar-SA","ar","Arabic",ChimeraDirection.Rtl) : new("en-US","en","Latin",ChimeraDirection.Ltr);
}

File: desktop/kotlin/ChimeraLocale.kt
package chimera.desktop

enum class Direction { LTR, RTL }
data class ChimeraLocale(val id: String, val language: String, val script: String, val direction: Direction) {
  companion object { fun forId(id: String?) = if (id?.startsWith("ar", true) == true) ChimeraLocale("ar-SA","ar","Arabic",Direction.RTL) else ChimeraLocale("en-US","en","Latin",Direction.LTR) }
}
